contracts/marketplace/Makefile.toml

```diff
@@ -0,0 +1,205 @@
+    soroban config identity generate trader_$TRADER
+    curl "${friendbot}?addr=$(soroban config identity address trader_$TRADER)"
+  done
+'''
+
+[tasks.accounts]
+dependencies = ["create_account_admin", "create_traders"]
+
+[tasks.get_token]
+script = '''
+soroban lab token id --asset native --network ${network} > /tmp/token_id.txt
+'''
+
+[tasks.get_asset]
+script = '''
+soroban lab token id --asset native --network ${network} > /tmp/asset_id.txt
+'''
+
+[tasks.deploy]
+script = '''
+  soroban contract deploy \
+  --source admin \
+  --wasm ../../target/wasm32-unknown-unknown/release/marketplace.wasm \
+  --network ${network} > /tmp/contract_id.txt
+'''
+
+[tasks.invoke_init]
+script = '''
+  soroban contract invoke \
+  --id $(cat /tmp/contract_id.txt) \
+  --source admin \
+  --network ${network} \
+  -- \
+  init \
+  --token $(cat /tmp/token_id.txt) \
+  --admin $(soroban config identity address admin) \
+  --percentage 1 \
+'''
+
+
+[tasks.invoke_create_listing]
+script = '''
+    soroban contract invoke \
+    --id $(cat /tmp/contract_id.txt) \
+    --source trader_1 \
+    --network ${network} \
+    -- \
+    create_listing \
+    --seller $(soroban config identity address trader_1)
+    --asset $(cat /tmp/asset_id.txt) \
+    --price 100 \
+'''
+
+[tasks.invoke_buy_listing]
+script = '''
+  soroban contract invoke \
+  --id $(cat /tmp/contract_id.txt) \
+  --source trader_2 \
+  --network ${network} \
+  -- \
+  buy_listing \
+  --buyer $(soroban config identity address trader_2)
+  --asset $(cat /tmp/asset_id.txt) \
+  --price 100 \
+'''
+
+[tasks.invoke_update_listing]
+script = '''
+  soroban contract invoke \
+  --id $(cat /tmp/contract_id.txt) \
+  --source trader_2 \
+  --network ${network} \
+  -- \
+  update_listing \
+  --random_seed 1234
+'''
+
+[tasks.invoke_pause_listing]
+script = '''
+  soroban contract invoke \
+  --id $(cat /tmp/contract_id.txt) \
+  --source admin \
+  --network ${network} \
+  -- \
+  play_raffle \
+  --random_seed 1234
+'''
+
+[tasks.invoke_remove_listing]
+script = '''
+  soroban contract invoke \
+  --id $(cat /tmp/contract_id.txt) \
+  --source admin \
+  --network ${network} \
+  -- \
+  play_raffle \
+  --random_seed 1234
+'''
+
+
+[tasks.invoke_contract]
+dependencies = ["invoke_init", "invoke_create_listing"]
+
+# currently this only displays balance of each player, without checking who has the most
+[tasks.get_balances]
+script = '''
+for TRADER in 1 2 3
+  do
+    soroban contract invoke \
+    --id $(cat /tmp/token_id.txt) \
+    --network ${network} \
+    -- \
+    balance \
+    --id $(soroban config identity address trader_$TRADER)
+  done
+'''
+
+[tasks.get_transactions]
+script = '''
+curl "${server}accounts/$(soroban config identity address trader_2)/transactions" > /tmp/transactions.txt
+'''
+
+[tasks.get_admin_transactions]
+script = '''
+curl "${server}accounts/$(soroban config identity address admin)/transactions" > /tmp/admin_transactions.txt
+'''
+
+[tasks.calculate_fees]
```

Comment:
In the future we should think about moving some of the common tasks into separate file. Practically every test will call `calculate_fees` and probably few more.